static/js/recent_senders.js

```diff
@@ -61,4 +62,37 @@ exports.compare_by_recency = function (user_a, user_b, stream_id, topic) {
     return 0;
 };
 
+exports.process_topic_edit = function (stream_id, old_topic, new_topic) {
+    // Since most topics don't have more than 10-20 messages, this shouldn't
+    // be much expensive.
+    const topic_dict = topic_senders.get(stream_id);
+    // We remove the old topic and process and new & old topics.
+    topic_dict.delete(old_topic);
+    topic_senders.set(stream_id, topic_dict);
+
+    const old_topic_msgs = util.get_messages_in_topic(stream_id, old_topic);
+    old_topic_msgs.forEach((msg) => {
+        exports.process_message_for_senders(msg);
+    });
+
+    const new_topic_msgs = util.get_messages_in_topic(stream_id, new_topic);
+    new_topic_msgs.forEach((msg) => {
+        exports.process_message_for_senders(msg);
+    });
+}
+
+exports.get_topic_recent_senders = function (stream_id, topic) {
+    const topic_dict = topic_senders.get(stream_id);
+    let sender_message_ids = new Map();
+    if (topic_dict !== undefined) {
+        sender_message_ids = topic_dict.get(topic);
+    }
+    const sorted_senders = Array.from(sender_message_ids.entries()).sort(
+        (s1, s2) => { return s1[1] - s2[1]; }
+    );
+    const recent_senders = [];
+    sorted_senders.forEach((item) => { recent_senders.push(item[0]); });
+    return recent_senders;
+};
```

Comment:
Can you add the `get_topic_recent_senders` change as a new commit on the other `recent_senders` data pipeline PR?  This seems like it could be self-contained and let us shrink this PR; it's clearly useful logic.